packages/events/test/register.test.ts

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2019 Environmental Systems Research Institute, Inc.
+ * Apache-2.0 */
+
+import { registerForEvent, unregisterForEvent } from "../src/register";
+
+import { encodeParam } from "@esri/arcgis-rest-request";
+import { UserSession } from "@esri/arcgis-rest-auth";
+
+export const TOMORROW = (function() {
+  const now = new Date();
+  now.setDate(now.getDate() + 1);
+  return now;
+})();
+
+import * as fetchMock from "fetch-mock";
+
+describe("groups", () => {
+  afterEach(fetchMock.restore);
+
+  describe("authenticted methods", () => {
+    const MOCK_REQOPTS = {
+      authentication: new UserSession({
+        clientId: "clientId",
+        redirectUri: "https://example-app.com/redirect-uri",
+        token: "fake-token",
+        tokenExpires: TOMORROW,
+        refreshToken: "refreshToken",
+        refreshTokenExpires: TOMORROW,
+        refreshTokenTTL: 1440,
+        username: "casey",
+        password: "123456",
+        portal: "https://myorg.maps.arcgis.com/sharing/rest"
+      })
+    };
+
+    it("should help a user register for an event", done => {
+      fetchMock.once("*", { success: true, groupId: "5bc" });
+
+      registerForEvent({ groupId: "5bc", ...MOCK_REQOPTS })
+        .then(() => {
+          expect(fetchMock.called()).toEqual(true);
+          const [url, options]: [string, RequestInit] = fetchMock.lastCall("*");
+          expect(url).toEqual(
+            "https://myorg.maps.arcgis.com/sharing/rest/community/groups/5bc/join"
```

Comment:
You're testing implementation details of rest-js. It would be better to stub `joinGroup()` and verify that it was called w/ the right args.

Reply:
good catch.

i initially skipped this out of laziness, but to be fair, we couldn't have spied until `joinGroup` and `leaveGroup` landed anyway.

Reply:
I figgered it was a 🐔 -> 🥚  situation